go_modules/lib/dependabot/go_modules/file_parser.rb

```diff
@@ -139,8 +139,9 @@ def go_mod_content
       def handle_parser_error(path, stderr)
         case stderr
         when /go: .*: unknown revision/m
-          line = stderr.lines.grep(/unknown revision/).first
-          raise Dependabot::DependencyFileNotResolvable, line.strip
+          line = stderr.lines.grep(/unknown revision/).first.strip
+          handle_github_unknown_revision(line) if line.start_with?("go: github.com/")
```

Comment:
Question: should this behaviour still be specific to `github.com/.*` urls?

Raising a typed exception for inaccessible module dependencies should be useful regardless of host, and the pattern seems portable.

I kept this filter for defensiveness.

Reply:
In theory they could be mercurial or bazaar deps?